internal/providers/pke/pkeworkflow/create_subnet_activity.go

```diff
@@ -0,0 +1,170 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkeworkflow
+
+import (
+	"context"
+	"io/ioutil"
+
+	"emperror.dev/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"go.uber.org/cadence/activity"
+
+	eksWorkflow "github.com/banzaicloud/pipeline/internal/cluster/distribution/eks/eksprovider/workflow"
+	internalAmazon "github.com/banzaicloud/pipeline/internal/providers/amazon"
+)
+
+const CreateSubnetActivityName = "pke-create-subnet"
+
+func NewCreateSubnetActivity(awsClientFactory *AWSClientFactory) *CreateSubnetActivity {
+	return &CreateSubnetActivity{
+		awsClientFactory: awsClientFactory,
+	}
+}
+
+// CreateSubnetActivity responsible for setting up a Subnet for an EKS cluster
+type CreateSubnetActivity struct {
+	awsClientFactory *AWSClientFactory
+}
+
+// CreateSubnetActivityInput holds data needed for setting up
+// a Subnet for EKS cluster
+type CreateSubnetActivityInput struct {
+	AWSActivityInput
+	ClusterID   uint
+	ClusterName string
+	// the ID of the VPC to create the subnet into
+	VpcID string
+
+	// the ID of the Route Table to associate the Subnet with
+	RouteTableID string
+
+	// The CIDR of the subnet
+	Cidr string
+
+	// The availability zone of the subnet
+	AvailabilityZone string
+}
+
+// CreateSubnetActivityOutput holds the output data of the CreateSubnetActivity
+type CreateSubnetActivityOutput struct {
+	SubnetID         string
+	Cidr             string
+	AvailabilityZone string
+}
+
+func (a *CreateSubnetActivity) Execute(ctx context.Context, input CreateSubnetActivityInput) (*CreateSubnetActivityOutput, error) {
+	logger := activity.GetLogger(ctx).Sugar().With(
+		"organization", input.OrganizationID,
+		"cluster", input.ClusterName,
+		"vpcID", input.VpcID,
+		"routeTableID", input.RouteTableID,
+		"subnetCidr", input.Cidr,
+		"availabilityZone", input.AvailabilityZone,
+		"secret", input.SecretID,
+	)
+
+	client, err := a.awsClientFactory.New(input.OrganizationID, input.SecretID, input.Region)
+	if err = errors.WrapIf(err, "failed to create AWS session"); err != nil {
+		return nil, err
+	}
+
+	buf, err := ioutil.ReadFile("templates/pke/subnet.cf.yaml")
```

Comment:
Probably we need to generalize `getEksCloudFormationTemplate` and reuse it here, see: https://github.com/banzaicloud/pipeline/blob/master/internal/cluster/distribution/eks/eksprovider/workflow/templates.go#L37